### DISCUSSION TITLE
moving forward with new features

### DISCUSSION THREAD

**Discussion by amtoine** [a2n-s/dotfiles#20] [Ideas]
this conversation is mainly for future us, as i'm writting it :relieved: 

i think there are a lot of things that can be improved on this configuration :thinking: 

## currently opened feature requests
- https://github.com/goatfiles/dotfiles/issues/12 :green_circle: 
- https://github.com/goatfiles/dotfiles/issues/14
- https://github.com/goatfiles/pkgbuilds/issues/22
- https://github.com/goatfiles/scripts/issues/37

### personal configuration
these dotfiles were designed for my personal use at the beginning
now that we are more than one, we need to seriously think about how to let the user change some parts of the configuration, e.g. the font or the shell used :grey_exclamation: 
we could get some inspiration from [scripts/apply-to-hunks](https://github.com/goatfiles/scripts/blob/main/scripts/amtoine-apply-to-hunks) which tries to centralize the theme management inside a single `dmenu` application :thinking: 
and we could very well extend that to:
- the themes obviously
- the fonts
- the terminal
- the shell
- the `git` configuration, e.g. the name and the email
- ...

what do you think @atxr? :yum:

**Comment by dolphinau** [comment]
Sounds good for me!
I'm currently working to finish #12 but we should definitely think about a *theme management* next!

**Reply by amtoine** [reply]
there is already something in [`goatfiles/scripts`](https://github.com/goatfiles/scripts) :muscle: 

work still has to be done, but that's not the priority i think, even though really cool, do not get me wrong :yum: 

all this is for the future us who will have addressed all the topics in https://github.com/orgs/goatfiles/discussions/19 :wink:

**Reply by amtoine** [reply]
in the same direction as the _theme management_ are some other things one might want to _**tweak**_:
- the **editor**: `helix`, `vim`, `nvim`, `emacs`?
- the **shell**:  `nushell`, `bash`, `fish`?

in the end, we need a custom _tweak tool_ :smirk:
